src/templates/cmpapi.js

```diff
@@ -164,7 +232,8 @@
 
         image = document.createElement('img');
         image.src =
-          '<%-CONSENT_SERVER_PROTOCOL%>://<%-CONSENT_SERVER_HOST%>/<%-API_VERSION%>/set-consent?consentByVendorId=' +
+          window.location.protocol +
+          '//<%-CONSENT_SERVER_HOST%>/<%-API_VERSION%>/set-consent?consentByVendorId=' +
           serializeConsentByVendorId(consentDecisionByVendorId) +
           (channelId !== '' ? '&channelId=' + channelId : '');
```

Comment:
The setting of consent is a bit inconsistent in my opinion .. the cookie is set when the image is retrieved but the local storage is updated immediately in the updateLocalStorageConsent() method on line 240. - Should this function not be called in the image.onload callback?